presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java

```diff
@@ -132,6 +136,46 @@ public PlanWithConsumedDynamicFilters visitJoin(JoinNode node, Set<String> allow
             return new PlanWithConsumedDynamicFilters(node, ImmutableSet.copyOf(consumed));
         }
 
+        @Override
+        public PlanWithConsumedDynamicFilters visitSemiJoin(SemiJoinNode node, Set<String> allowedDynamicFilterIds)
```

Comment:
Can you extract common code with `visiJoin`?

Reply:
I feel it's not very straightforward to do so.